### PR TITLE
LEA on rhs of block copy can't be contained

### PR DESCRIPTION
**Pull request by CarolEidt**
We don't usually create an LEA on the rhs of a block copy - we check the type of the indir, and if it's struct we avoid the copy. However, in this case, the rhs was the address of a scalar field within a struct, so the indir was not TYP_STRUCT.

### CODE REVIEW

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib PTAL

File: src/jit/lowerarmarch.cpp [hunk 1]
```diff
@@ -382,6 +382,11 @@ void Lowering::LowerBlockStore(GenTreeBlk* blkNode)
         if (source->gtOper == GT_IND)
         {
             MakeSrcContained(blkNode, source);
+            GenTree* addr = source->AsIndir()->Addr();
+            if (!addr->OperIsLocalAddr())
+            {
+                addr->ClearContained();
```
Comment (sandreenko) [inline src/jit/lowerarmarch.cpp]: Does `MakeSrcContained` set contained for the `addr'? Why can't it check that in this case it should not set it?

Comment (CarolEidt) [reply src/jit/lowerarmarch.cpp]: `MakeSrcContained` only sets `source` as contained. It doesn't check any of the children of `source` - and in some cases it is perfectly valid for an indir to be contained as well as a child `LEA`.

Comment (CarolEidt) [reply src/jit/lowerarmarch.cpp]: Also note that this same code was already here: https://github.com/dotnet/coreclr/blob/master/src/jit/lowerxarch.cpp#L383

**Comment by CarolEidt** [comment]
@dotnet/jit-contrib ping

**Pull request merged**